napari/layers/utils/stack_utils.py

```diff
@@ -0,0 +1,103 @@
+import warnings
+from typing import Union, List
+
+import numpy as np
+from napari.layers import Image, Labels
+from vispy.color import Colormap
+
+
+def stack_to_images(
+    stack: Union[Image, Labels],
+    axis: int,
+    colormaps: List[Union[str, Colormap]] = None,
+) -> List[Image]:
+    """Function to split the active layer into separate layers along an axis
+
+    Parameters
+    ----------
+    stack : napari.layers.Image
+        The image stack to be split into a list of image layers
+    axis : int
+        The axis to split along.
+
+    Returns
+    -------
+    list
+        List of images
+    """
+
+    data = stack.data
+    name = stack.name
+    num_dim = len(data.shape)
+
+    if num_dim < 3:
+        warnings.warn(
+            "The image needs more than 2 dimensions for splitting", UserWarning
+        )
+        return None
+
+    if axis >= num_dim:
+        warnings.warn(
+            "Can't split along axis {}. The image has {} dimensions".format(
+                axis, num_dim
+            ),
+            UserWarning,
+        )
+        return None
+
+    cmap = stack.colormap
+    if colormaps is None:
+        colormaps = data.shape[axis] * [cmap]
```

Comment:
Sorry to restart this discussion, but I think when splitting a stack, we should follow the convention of `add_image(..., channel_axis=ax)`. This is handled here:

https://github.com/napari/napari/blob/b94ed99b8f60599aba61579f5711c8bf6e8896d2/napari/components/add_layers_mixin.py#L236-L247

We can definitely argue that choice and change it if we decide it was the wrong choice, but we haven't had any complaints, and either way the behaviours should match, preferably even through shared code.